LibNoDaveConnectionLibrary/DataTypes/Blocks/Step7V5/S7FunctionBlockRow.cs

```diff
@@ -621,40 +622,36 @@ public IEnumerable<String> Dependencies
                 }
                 else if (Command == Mnemonic.opCALL[(int)MnemonicLanguage])
                 {
+                    var deps = new List<String>();
+
                     if (!String.IsNullOrEmpty(DiName))
                     {
-                        /*Block blk1, blk2;
-                        var fld = (this.Parent).ParentFolder as BlocksOfflineFolder;
-                        if (fld != null && !Parameter.StartsWith("#"))
-                        {
-                            var blk1 = fld.GetBlock(Parameter);                           
-                        }
+                        var db = DiName.Replace("DI", "DB");
 
-                        return new List<Block>() { blk1, blk2 };*/
+                        deps.Add(db);
                     }
-                    else
-                    {
-                        var fld = (this.Parent).ParentFolder as BlocksOfflineFolder;
-                        if (fld != null && !Parameter.StartsWith("#"))
-                        {
-                            //var blk = fld.GetBlock(Parameter);
 
-                            return new List<String>() { Parameter };
-                        }
+                    var fld = (this.Parent).ParentFolder as BlocksOfflineFolder;
+
+                    if (fld != null && !Parameter.StartsWith("#"))
+                    {
+                        deps.Add(Parameter);
                     }
+
+                    return deps;
                 }
                 else if (Parameter.StartsWith("DB") && Parameter[2] != '[' && Parameter[2] != 'D' && Parameter[2] != 'W' && Parameter[2] != 'B' && Parameter[2] != 'X' && this.Parent != null)
                 {
                     var paras = Parameter.Split(new[] { '.' });
                     var fld = (this.Parent).ParentFolder as BlocksOfflineFolder;
                     if (fld != null)
                     {
-                        if (paras.Length > 1)
+                        if (paras.Length > 0)
                         {
-                            var byteAdr = int.Parse(paras[1].Replace("DBX", "").Replace("DBB", "").Replace("DBW", "").Replace("DBD", ""));
+                            // var byteAdr = int.Parse(paras[1].Replace("DBX", "").Replace("DBB", "").Replace("DBW", "").Replace("DBD", ""));
 
-                            var bitAdr = 0;
-                            if (paras.Length > 2) bitAdr = int.Parse(paras[2]);
+                            // var bitAdr = 0;
+                            // if (paras.Length > 2) bitAdr = int.Parse(paras[2]);

```

Comment:
this code wasn't being used and was causing an error on a few data blocks so just commenting it for now

also checking .Length > 0 since we're only accessing index 0 in paras